examples/train_classification_model.py

```diff
@@ -0,0 +1,199 @@
+wandb_logger = WandbLogger(
+    project="example-script",
+    entity="graphnet-team",
+    save_dir=WANDB_DIR,
+    log_model=True,
+)
+
+
+def train(config: Dict[str, Any]) -> None:
+    """Train model with configuration given by `config`."""
+    # Log configuration to W&B
+    wandb_logger.experiment.config.update(config)
+
+    # Common variables; equal distribution of pid.
+    train_selection = get_desired_event_numbers(
+        database=cast(str, config["db"]),
+        desired_size=cast(int, config["event_numbers"]),
+        fraction_noise=float(1 / 3),
+        fraction_muon=float(1 / 3),
+        fraction_nu_e=float(1 / 9),
+        fraction_nu_mu=float(1 / 9),
+        fraction_nu_tau=float(1 / 9),
+    )
+
+    logger.info(f"features: {features}")
+    logger.info(f"truth: {truth}")
+
+    (
+        training_dataloader,
+        validation_dataloader,
+    ) = make_train_validation_dataloader(
+        cast(str, config["db"]),
+        train_selection,
+        cast(str, config["pulsemap"]),
+        features,
+        truth,
+        batch_size=cast(int, config["batch_size"]),
+        num_workers=cast(int, config["num_workers"]),
+    )
+
+    # Building model
+    detector = IceCubeDeepCore(
+        graph_builder=KNNGraphBuilder(nb_nearest_neighbours=8),
+    )
+    gnn = DynEdge(
+        nb_inputs=detector.nb_outputs,
+        global_pooling_schemes=["min", "max", "mean", "sum"],
+    )
+    task = ClassificationTask(
+        nb_classes=len(np.unique(list(config["class_options"].values()))),
+        hidden_size=gnn.nb_outputs,
+        target_labels=config["target"],
+        loss_function=CrossEntropyLoss(options=config["class_options"]),
+        transform_inference=lambda x: softmax(x, dim=-1),
+    )
+    model = StandardModel(
+        detector=detector,
+        gnn=gnn,
+        tasks=[task],
+        optimizer_class=Adam,
+        optimizer_kwargs={"lr": 1e-04, "eps": 1e-03},
+    )
+
+    # Training model
+    callbacks = [
+        EarlyStopping(
+            monitor="val_loss",
+            patience=config["patience"],
+        ),
+        ProgressBar(),
+    ]
+
+    trainer = Trainer(
+        accelerator=config["accelerator"],
+        devices=config["devices"],
+        max_epochs=config["n_epochs"],
+        callbacks=callbacks,
+        log_every_n_steps=1,
+        logger=wandb_logger,
+    )
+
+    try:
+        trainer.fit(model, training_dataloader, validation_dataloader)
+    except KeyboardInterrupt:
+        logger.warning("[ctrl+c] Exiting gracefully.")
+        pass
+
+    # Saving predictions to file
+    results = get_predictions(
+        trainer,
+        model,
+        validation_dataloader,
+        [
+            cast(str, config["target"]) + "_noise_pred",
+            cast(str, config["target"]) + "_muon_pred",
+            cast(str, config["target"]) + "_neutrino_pred",
+        ],
+        additional_attributes=[cast(str, config["target"]), "event_no"],
+    )
```

Comment:
I would suggest you use the new `Model.predict{,_as_dataframe}(...)` method(s).

Reply:
added a prediction segment, but need help setting it up.